src/mod/gcewing/sg/util/GateUtil.java

```diff
@@ -2,20 +2,41 @@
 
 import gcewing.sg.tileentity.DHDTE;
 import gcewing.sg.tileentity.SGBaseTE;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
 public class GateUtil {
+
+    public static SGBaseTE findGate(World world, EntityPlayer player, int radius) {
```

Comment:
This should not require a player. Should take the blockpos.

Reply:
findGate is never used with a BlockPos starting point, it always comes from a player. Everytime, the blockPos had to be constructed from the player.
I would have provided the player param as an overload if at least one place in the code started from a block pos

Reply:
That doesn't matter. It does not use the Player at all BESIDES getting the block pos.